### PR TITLE
Alternative way of creating decoder_attention_mask to avoid torch warning about new_tensor()

### PR DESCRIPTION
**Pull request by kmartiny**
# What does this PR do?

Replace the method of creating `decoder_attention_mask` in `modeling_encoder_decoder.py` to avoid a pytorch warning. Masks were created as 
```python
decoder_attention_mask = torch.tensor(decoder_input_ids != self.config.pad_token_id)
```
but this results in a torch warning:
> transformers/models/encoder_decoder/modeling_encoder_decoder.py:620: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than tensor.new_tensor(sourceTensor).
  decoder_attention_mask = decoder_input_ids.new_tensor(decoder_input_ids != self.config.pad_token_id)

Creating the masks through
```python
decoder_attention_mask = torch.not_equal(decoder_input_ids, self.config.pad_token_id)
```
is equivalent without producing warnings.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Looks alright to me. Do you mind rebasing to main and making sure the CI is green! 🤗 might be a flaky test in this case.
